samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -26,12 +28,16 @@
 LOG = logging.getLogger(__name__)
 FUNCTION_NAME = "FunctionName"
 PAYLOAD = "Payload"
+EVENT_STREAM = "EventStream"
+PAYLOAD_CHUNK = "PayloadChunk"
+INVOKE_COMPLETE = "InvokeComplete"
+LOG_RESULT = "LogResult"
 
 
-class LambdaInvokeExecutor(BotoActionExecutor):
+class AbstractLambdaInvokeExecutor(BotoActionExecutor, ABC):
     """
-    Calls "invoke" method of "lambda" service with given input.
-    If a file location provided, the file handle will be passed as Payload object
+    Abstract class for different lambda invocation executors, see implementation for details.
+    For Payload parameter, if a file location provided, the file handle will be passed as Payload object
     """
 
     _lambda_client: Any
```

Comment:
I've tried setting its type to `LambdaClient` but then I've faced following issues;
- First is the parameters. What we are doing today is to send some pre-defined and some arbitrary ones that can be set by customers. We are relying on boto3 errors for the case customer provides an invalid option for not keeping source of truth in our codebase. But if I set client to `LambdaClient` then I can't keep using decomposing a dictionary to use as parameters. I need to extract them one by one to use in function call since there is no type definition for the function parameters.
- Second is the response,  in mypy it has typed dictionary but mypy somehow doesn't like that to be returned by `Dict[Any, Any]` even though it should cover wider. 

We will have a refactoring task next week, we will try to re-introduce typing there.